substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1990,6 +1992,23 @@ pub mod pallet {
 			}
 			Ok(Some(T::WeightInfo::deprecate_controller_batch(controllers.len() as u32)).into())
 		}
+
+		/// TODO: docs
+		/// Wrapper extrinsic for `do_delegated_bond`, but accessible from outside the current
+		/// context, guarded by `DelegateOrigin`.
+		#[pallet::call_index(29)]
+		#[pallet::weight(T::WeightInfo::bond())]
+		pub fn delegated_bond(
+			origin: OriginFor<T>,
+			stash: T::AccountId,
+			#[pallet::compact] value: BalanceOf<T>,
+			payee: RewardDestination<T::AccountId>,
+		) -> DispatchResult {
+			// `DelegateOrigin` type lets approved pallets on different parachains to call into this
+			// privileged extrinsic.
+			T::DelegateOrigin::ensure_origin(origin)?;
```

Comment:
How do we verify the `value` is available to bond (delegated and unbonded)?

Reply:
Why do we need to verify it if it's coming from a trusted delegated staking source? It should have already been verified by the caller.

But technically if we'll need it (though I think right now we don't), we can just make `DelegatorOrigin` return the available balance on the delegate side on success.